### ISSUE TITLE
pipe-operator support for Nix

### ISSUE THREAD

**Issue by viperML** [nvim-treesitter/nvim-treesitter#7383]
### Describe the highlighting problem

As https://github.com/nix-community/tree-sitter-nix is radio silent, I will try to open the issue here.

New operators `<|`, `|>` were merged into Nix, gated by the `pipe-operator experimental feature`. These are not yet supported by tree-sitter.

Links:
- Nix PR: https://github.com/NixOS/nix/pull/11131
- PR to tree-sitter-nix (stale?): https://github.com/nix-community/tree-sitter-nix/pull/85

Pinging some people that contributed to the Nix stuff in nvim-treesitter @figsoda @leo60228 @stephen-huan @cstrahan

### Example snippet that causes the problem

```nix
[ 1 2 ] |> builtins.head
```

### Tree-sitter parsing result

```
(source_code ; [0, 0] - [1, 0]
  expression: (binary_expression ; [0, 0] - [0, 22]
    left: (list_expression ; [0, 0] - [0, 5]
      element: (integer_expression) ; [0, 1] - [0, 2]
      element: (integer_expression)) ; [0, 3] - [0, 4]
    (ERROR ; [0, 6] - [0, 7]
      (ERROR)) ; [0, 6] - [0, 7]
    right: (select_expression ; [0, 9] - [0, 22]
      expression: (variable_expression ; [0, 9] - [0, 17]
        name: (identifier)) ; [0, 9] - [0, 17]
      attrpath: (attrpath ; [0, 18] - [0, 22]
        attr: (identifier))))) ; [0, 18] - [0, 22]
```

### Example screenshot

![Image](https://github.com/user-attachments/assets/87a80f5f-2024-43a4-a0ae-680a413b712e)

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- WARNING `node` executable not found (only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `git` executable found.
- ERROR `cc` executable not found.
  - ADVICE:
    - Check that any of { vim.NIL, "cc", "gcc", "clang", "cl", "zig" } is in your $PATH or set the environment variable CC or `require"nvim-treesitter.install".compilers` explicitly!
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.11.4",
  sysname = "Linux",
  version = "#1-NixOS SMP PREEMPT_DYNAMIC Thu Oct 17 13:27:02 UTC 2024"
} ~

Parser/Features         H L F I J
  - asm                 ✓ . . . ✓
  - astro               ✓ ✓ ✓ ✓ ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - cmake               ✓ . ✓ ✓ ✓
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - elixir              ✓ ✓ ✓ ✓ ✓
  - go                  ✓ ✓ ✓ ✓ ✓
  - haskell             ✓ ✓ ✓ . ✓
  - hcl                 ✓ . ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - meson               ✓ . ✓ ✓ ✓
  - nix                 ✓ ✓ ✓ ✓ ✓
  - org                 . . . . .
  - query               ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - scheme              ✓ . ✓ . ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - typst               ✓ . ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1713773202
Run "nvim -V1 -v" for more info
```

**Comment by clason** [comment]
PR welcome; all queries are user-contributed.
